transaction.go

```diff
@@ -198,7 +197,18 @@ type PackedTransaction struct {
 
 func (p *PackedTransaction) ID() SHA256Bytes {
 	h := sha256.New()
-	_, _ = h.Write(p.PackedTransaction)
+
+	switch p.Compression {
+	case CompressionNone:
+		_, _ = h.Write(p.PackedTransaction)
```

Comment:
why not `h.Write(p.PackedTransaction)`?

Reply:
just add a case: if tx is packed with zlib_compression, then it should be decompressed in ID function